### PR TITLE
Blog app - controllers specs

### PR DESCRIPTION
**Pull request by TesheMaximillan**
**Hello Reviewer** 👋

> In this milestone, I set up test cases for the `UsersController` and `PostsController`.

### Basic Requirements:🟢
***
* Add RSpec as a gem to the project.
* Create a Request Spec for the `UsersController` and `PostsController`.
* The following test case is checked:
  * if response was successful, 
  * if a correct template was rendered.
  * if the response body includes correct placeholder text.

### CODE REVIEW

File: spec/requests/posts_request_spec.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
+require 'rails_helper'
+
```
Comment (thecodechaser) [inline spec/requests/posts_request_spec.rb:2]: - [ ] You currently have 4 out of 8 tests failing. Please take a look below and fix them.

image

Comment (thecodechaser) [inline spec/requests/posts_request_spec.rb:2]: - [ ] Your tests are missing to check `If a correct template was rendered`. Please make sure to include this check for all of your controllers.
  - Hint:  `it 'should render a template' do expect(response).to render_template('something') end`

**Review by thecodechaser — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ 

## Hi @TesheMaximillan👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 🎉
- No linter errors. :heavy_check_mark:
- Gitflow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- PR has a good title and summary. ✔️



## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/requests/posts_request_spec.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  describe 'GET /users/:user_id/posts/:id' do
+    it 'check the status is successful' do
+      get user_post_path(1, 1)
+      expect(response).to have_http_status(200)
+    end
+
+    it 'check the body has a placeholder' do
+      get user_post_path(1, 1)
+      expect(response.body).to include('Here is a single post with a collection of comments')
+    end
```
Comment (TesheMaximillan) [inline spec/requests/posts_request_spec.rb:30]: This is how I tested if  a correct template was rendered

File: spec/requests/posts_request_spec.rb (hunk 1 above)
Comment (TesheMaximillan) [reply spec/requests/posts_request_spec.rb:2]: rspec
test for controllers

Comment (TesheMaximillan) [reply spec/requests/posts_request_spec.rb:2]: As you see, all the tests are passed.

Comment (TesheMaximillan) [reply spec/requests/posts_request_spec.rb:2]: test for controllers
rspec

Comment (TesheMaximillan) [reply spec/requests/posts_request_spec.rb:2]: As you see all of the tests are passed in my environment

**Comment by TesheMaximillan** [comment]
**Dear reviewer 👋**

* I tried to see the possible errors and why the tests failed on your side, so, please try to install the bundle package, maybe that was the error, or re-run the test again
  
N.B. All tests are passed on my side

**Review by uwadonat — APPROVED** [review]
## Hi @TesheMaximillan,

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
## Congratulations! 🎉 

# Status: Approved ✔️✔️✔️

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100%.
Please, remember to tag me @uwadonat in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Pull request merged**